test/src/expectEvent.truffle.test.js

```diff
@@ -463,12 +523,161 @@ contract('expectEvent (truffle contracts)', function ([deployer]) {
+          this.normalValue2,
+          this.secondEmitter.address
+        );
+        this.txHash = receipt.transactionHash;
+      });
+
+      context('with directly called contract', function () {
+        it('accepts emitted events with correct indexed parameter and emitter object', async function () {
+          expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedUint', {
+            indexedValue: this.indexedValue,
+            normalValue: this.normalValue,
+          });
         });
+
+        it('accepts emitted indexed events with correct indexed parameter and emitter class', async function () {
+          expectEvent.inTransaction(this.txHash, EventEmitter, 'IndexedUint', {
+            indexedValue: this.indexedValue,
+            normalValue: this.normalValue,
+          });
+        });
+      });
+
+      context('with indirectly called contract', function () {
+        it('accepts events emitted from other contracts with emitter object', async function () {
+          expectEvent.inTransaction(this.txHash, this.secondEmitter, 'IndexedUint', {
+            indexedValue: this.indexedValue2,
+            normalValue: this.normalValue2,
+          });
+        });
+
+        it('accepts emitted indexed events with emitter class', async function () {
+          expectEvent.inTransaction(this.txHash, IndirectEventEmitter, 'IndexedUint', {
+            indexedValue: this.indexedValue2,
+            normalValue: this.normalValue2,
+          });
+        });
+      });
+    });
+
+    describe('with conflicting indexed event parameters', function () {
+      beforeEach(async function () {
+        this.indexedValue = new BN(42);
+        this.normalValue = new BN(2014);
+        this.indexedConflictValue = new BN(2016);
+        const { receipt } = await this.emitter.emitIndexedConflictingUint(
+          this.indexedValue,
+          this.normalValue,
+          this.indexedConflictValue,
+          this.secondEmitter.address
+        );
+        this.txHash = receipt.transactionHash;
+      });
+
+      context('with directly called contract', function () {
+        it('accepts emitted events with correct indexed parameter', async function () {
+          await expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedConflictingUint', {
+            indexedValue: this.indexedValue,
+            normalValue: this.normalValue,
+          });
+        });
+
+        it('throws if the emitter class is passed', async function () {
+          await assertFailure(expectEvent.inTransaction(this.txHash, EventEmitter, 'IndexedConflictingUint', {
+            indexedValue: this.indexedValue,
+            normalValue: this.normalValue,
+          }));
+        });
+
+        it('throws if the event value emitted from other contract is passed', async function () {
+          await assertFailure(expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedConflictingUint', {
+            indexedValue: this.normalValue,
```

Comment:
This is why the parameters names of the indexed event are different, so I can do this type of checks.